gnomad/utils/constraint.py

```diff
@@ -125,3 +127,274 @@ def collapse_strand(
+                    ht[observed_variants][i] / ht[possible_variants],
+                    [1, ht[mu_snp]],
+                    weight=ht[possible_variants] if weighted else None,
+                ).beta,
+            )
+            for i in range(length)
+        ]
+    }
+    return ht.aggregate(hl.struct(**agg_expr))
+
+
+def build_plateau_models_total(
+    ht: hl.Table,
+    cpg: str,
+    mu_snp: str,
+    observed_variants: str,
+    possible_variants: str,
+    weighted: bool = False,
+) -> hl.Struct:
+    """
+    Build plateau model for all release samples without downsamplings to calibrate mutation rate to compute predicted proportion observed value.
+
+    The x and y of the plateau models:
+    - x: `mu_snp` - mutation rate
+    - y: proportion observed ('observed_variants'/ 'possible_variants')
+
+    :param ht: High coverage Table.
+    :param cpg: The annotation name of booleans that determine whether the site is CPG site or not.
+    :param mu_snp: The annotation name of the mutation rate.
+    :param observed_variants: The annotation name of the observed variant counts for each combination of keys in `ht`.
+    :param possible_variants: The annotation name of the possible variant counts for each combination of keys in `ht`.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants', defaults to False.
+    :return: A Dictionary of intercepts and slopes for a plateau model.
+    """
+    agg_expr = {
+        "total": hl.agg.group_by(
+            ht[cpg],
+            hl.agg.linreg(
+                ht[observed_variants] / ht[possible_variants],
+                [1, ht[mu_snp]],
+                weight=ht[possible_variants] if weighted else None,
+            ).beta,
+        )
+    }
+    return ht.aggregate(hl.struct(**agg_expr))
+
+
+def build_coverage_model(
+    ht: hl.Table,
+    low_coverage_obs_exp: str,
+    log_coverage: str,
+) -> Tuple[float, float]:
+    """
+    Build coverage model.
+
+    This function uses linear regression to build a model of log10(coverage) to this scaled ratio as a correction
+    factor for predicted proportion observed/expected variant counts at low coverage sites.
+
+    The x and y of the coverage model:
+    - x: log10('exome_coverage') at low coverage site
+    - y: sum('observed_variants')/ (`high_coverage_scale_factor` * sum('possible_variants' * 'mu_snp') at low coverage site
+    where `high_coverage_scale_factor` = sum('observed_variants') / sum('possible_variants' * 'mu_snp') at high coverage site
+
+    :param ht: Low coverage Table.
+    :param low_cov_oe_expr: The annotation name of observed:expected ratio for a given coverage level.
+    :param log_cov_expr: The annotation name of log10 coverage.
+    :return: Tuple with intercept and slope of the model.
+    """
+    return tuple(
+        ht.aggregate(
+            hl.agg.linreg(ht[low_coverage_obs_exp], [1, ht[log_coverage]])
+        ).beta
+    )
```

Comment:
maybe this function doesn't actually need to accept a Table and could instead accept expressions, something like:
```suggestion
    return tuple(
        ht.aggregate(
            hl.agg.linreg(low_cov_oe_expr, [1, log_cov_expr)
        ).beta
    )
```